### PR TITLE
Activity API + RoutingTargets

### PR DESCRIPTION
**Pull request by ajungg**

### CODE REVIEW

File: ui/src/main/java/io/snabble/sdk/ui/payment/PaymentOptionsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import io.snabble.sdk.ui.R
+
+open class PaymentOptionsFragment : Fragment() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+    }
+
```
Comment (rekire) [inline ui/src/main/java/io/snabble/sdk/ui/payment/PaymentOptionsFragment.kt:14]: Useless code

File: ui/src/main/java/io/snabble/sdk/ui/payment/Payone.kt [hunk 2]
```diff
@@ -74,7 +73,7 @@ object Payone {
 
         val url = descriptor.links?.get("tokenization")
         if (url == null) {
-            project.events.logError("Datatrans Error: No tokenization url")
+            project.events.logError("Payone Error: No tokenization url")
```
Comment (rekire) [inline ui/src/main/java/io/snabble/sdk/ui/payment/Payone.kt:76]: Good catch!

File: ui/src/main/java/io/snabble/sdk/ui/payment/ProjectPaymentOptionsActivity.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package io.snabble.sdk.ui.payment
+
+import androidx.fragment.app.Fragment
+import io.snabble.sdk.ui.BaseFragmentActivity
+
+class ProjectPaymentOptionsActivity : BaseFragmentActivity() {
+    companion object {
+        const val ARG_BRAND = ProjectPaymentOptionsView.ARG_BRAND
```
Comment (rekire) [inline ui/src/main/java/io/snabble/sdk/ui/payment/ProjectPaymentOptionsActivity.kt:8]: Don't we need the `@JvmStatic` for all those kind of constants for Java compatibility?

File: ui/src/main/java/io/snabble/sdk/ui/search/SearchHelper.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package io.snabble.sdk.ui.search
+
+object SearchHelper {
+    var lastSearch: String? = null
+}
```
Comment (rekire) [inline ui/src/main/java/io/snabble/sdk/ui/search/SearchHelper.kt:5]: Might be a better idea to move this to the SnabbleUI class to eliminate this singleton.

File: ui/src/main/res/layout/snabble_item_searchable_product.xml [hunk 5]
```diff
@@ -4,6 +4,7 @@
     android:orientation="vertical"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
+    android:minHeight="48dp"
```
Comment (rekire) [inline ui/src/main/res/layout/snabble_item_searchable_product.xml:7]: Can you explain this change?

**Review by rekire — CHANGES_REQUESTED** [review]
I still need to test everything on the device, but this looks very solid.

File: ui/src/main/res/layout/snabble_item_searchable_product.xml (hunk 5 above)
Comment (ajungg) [reply ui/src/main/res/layout/snabble_item_searchable_product.xml:7]: Bothered me, search entries were uneven when no barcodes were present

File: ui/src/main/java/io/snabble/sdk/ui/payment/ProjectPaymentOptionsActivity.kt (hunk 3 above)
Comment (ajungg) [reply ui/src/main/java/io/snabble/sdk/ui/payment/ProjectPaymentOptionsActivity.kt:8]: const vals are already inplicit jvm static

**Review by rekire — CHANGES_REQUESTED** [review]
In the Aldi Project I see with an empty cart the payment selector.

**Pull request merged**